packet/conn.go

```diff
@@ -175,9 +187,29 @@ func (c *Conn) copyN(dst io.Writer, src io.Reader, n int64) (written int64, err
 		}
 		buf := c.copyNBuf[:bcap]
 
-		rd, err := io.ReadAtLeast(src, buf, bcap)
+		var rd int
+		rd, err = io.ReadAtLeast(src, buf, bcap)
+
 		n -= int64(rd)
 
+		// if we've read to EOF, and we have compression then advance the sequence number
+		// and reset the compressed reader to continue reading the remaining bytes
+		// in the next compressed packet.
+		if c.Compression != MYSQL_COMPRESS_NONE && rd < bcap &&
+			(goErrors.Is(err, io.ErrUnexpectedEOF) || goErrors.Is(err, io.EOF)) {
+			// we have read to EOF and read an incomplete uncompressed packet
+			// so advance the compressed sequence number and reset the compressed reader
+			// to get the remaining unread uncompressed bytes from the next compressed packet.
+			c.CompressedSequence++
+			if c.compressedReader, err = c.newCompressedPacketReader(); err != nil {
+				return written, errors.Trace(err)
+			}
+
+			// now read the remaining bytes into the buffer containing the first read bytes
+			rd, err = io.ReadAtLeast(c.currentPacketReader(), buf[rd:], bcap-rd)
```

Comment:
should we delete this reading and let the outer loop read it? because it may still meet the EOF error like line 191.

Reply:
Yes terrific suggestion, i've applied this change.